File: Examen1/2026_1/Resolucion/ExamenParcialCSharp/GestionAlumnosPersistance/Alumnos/DAO/AlumnoDAO.cs
using GestionAlumnosModel.Alumno;
using GestionAlumnosPersistance.DAO;

namespace GestionAlumnosPersistance.Alumnos.DAO
{
    public interface AlumnoDAO : IDAO<Alumno>
    {
        // Patron SELECT INTO: busca por codigo y retorna el id (0 si no existe)
        int BuscarPorCodigo(string codigo);
    }
}

File: lab11Calificado/Lab11_PLANTILLA/TestSoftSolution/TestSoftPersistance/src/main/java/pe/edu/pucp/testsoft/persistance/alumno/DAO/AlumnoDAO.java
package pe.edu.pucp.testsoft.persistance.alumno.DAO;

import pe.edu.pucp.testsoft.model.Alumno;
import pe.edu.pucp.testsoft.persistance.dao.IDAO;
import java.util.List;

public interface AlumnoDAO extends IDAO<Alumno> {
    int buscarPorCodigo(String codigo);              // patrón SELECT INTO: devuelve id (0 si no existe)
    List<Alumno> buscarPorNombreApellido(String texto);
}
